packages/roosterjs-plugin-picker/lib/PickerPlugin.ts

```diff
@@ -123,34 +120,34 @@ export default class PickerPlugin<T extends PickerDataProvider = PickerDataProvi
      * @param event PluginEvent object
      */
     public onPluginEvent(event: PluginEvent) {
-        if (
-            event.eventType == PluginEventType.ContentChanged &&
-            event.source == ChangeSource.SetContent &&
-            this.dataProvider.onContentChanged
-        ) {
-            // Undos and other major changes to document content fire this type of event.
-            // Inform the data provider of the current picker placed elements in the body.
-            let elementIds: string[] = [];
-            this.editor.queryElements(
-                "[id^='" + this.pickerOptions.elementIdPrefix + "']",
-                element => {
-                    if (element.id) {
-                        elementIds.push(element.id);
+        if (event.eventType == PluginEventType.ContentChanged) {
+            if (event.source == ChangeSource.SetContent && this.dataProvider.onContentChanged) {
+                // Undos and other major changes to document content fire this type of event.
+                // Inform the data provider of the current picker placed elements in the body.
+                let elementIds: string[] = [];
+                this.editor.queryElements(
+                    "[id^='" + this.pickerOptions.elementIdPrefix + "']",
+                    element => {
+                        if (element.id) {
+                            elementIds.push(element.id);
+                        }
                     }
+                );
+                this.dataProvider.onContentChanged(elementIds);
+            } else if (event.source == ChangeSource.CustomReplace) {
+                // Emoji shortcut autocomplete is handled by a separate plugin, CustomReplace
+                // When CustomReplace performs autocomplete, we can stop suggesting
+                if (this.isSuggesting) {
+                    this.setIsSuggesting(false);

```

Comment:
I don't like this idea. When suggesting, all other plugins should not handle keyboard event, including input event. Here it makes this logic mess up. A plugin should not know this kind of knowledge from another plugin.

We should still let emoji plugin handle the shortcuts starts with ":" otherwise picker plugin, as a common control, has too much knowledge about emoji.

Reply:
So you think CustomReplace should only handle emoji shortcuts that don't begin with ":"?

Reply:
From roosterjs layer, yes.

If you want CustomReplace also handle ":" emojis, you should override it in OWA rather handling it in roosterjs. See my comment in the PR of OWA.